updates to opencover module to make it more user friendly and works on coreclr

SteveL-MSFT opened the pull request:
added type to returned objects
replaced expand-ziparchive with call to Archive module
added defaults to location of opencover.xml, opencover installation, powershell.exe, and location of tests
updated error messages with corrective action
replaced System.Net.WebClient call (not available on coreclr) with invoke-webrequest
scoped change of executionpolicy to process when starting pester to avoid error if system policy is less restrictive
updated version of module

SteveL-MSFT committed:
added type to returned objects
replaced expand-ziparchive with call to Archive module
added defaults to location of opencover.xml, opencover installation, powershell.exe, and location of tests
updated error messages with corrective action
replaced System.Net.WebClient call (not available on coreclr) with invoke-webrequest
scoped change of executionpolicy to process when starting pester to avoid error if system policy is less restrictive
updated version of module

SteveL-MSFT commented:
cc @adityapatwardhan

adityapatwardhan commented on test/tools/OpenCover/OpenCover.psm1:
```diff
@@ -1,5 +1,7 @@
 ﻿#region privateFunctions
 
+$script:psRepoPath = git rev-parse --show-toplevel
```
>git rev-parse --show-toplevel [](start = 21, length = 29)

git tools are not available in Azure Functions or other automation pipelines. Please check if git tools exist before using them.

SteveL-MSFT replied:
adding check for git

adityapatwardhan commented on test/tools/OpenCover/OpenCover.psm1:
```diff
@@ -359,22 +332,15 @@ function Invoke-OpenCover
-        [parameter(Mandatory=$true)]$PowerShellExeDirectory,
+        [parameter()]$TestDirectory = "$($script:psRepoPath)/test/powershell",
+        [parameter()]$OpenCoverPath = "~/OpenCover",
+        [parameter()]$PowerShellExeDirectory = "$($script:psRepoPath)/src/powershell-win-core/bin/debug/netcoreapp1.0/win10-x64",
```
>"$($script:psRepoPath)/src/powershell-win-core/bin/debug/netcoreapp1.0/win10-x64", [](start = 47, length = 82)

This should be "$($script:psRepoPath)/src/powershell-win-core/bin/CodeCoverage/netcoreapp1.0/win10-x64"

SteveL-MSFT replied:
will fix

SteveL-MSFT commented on test/tools/OpenCover/OpenCover.psm1:
```diff
@@ -359,22 +332,15 @@ function Invoke-OpenCover
 
-    if(-not $PSBoundParameters.ContainsKey('OpenCoverPath'))
-    {        
-        $openCoverBin = (Get-Command -Name 'opencover.console' -ErrorAction Ignore).Source         
```
I'll add a different check if opencover.console is in the path

adityapatwardhan commented on test/tools/OpenCover/OpenCover.psm1:
```diff
@@ -332,18 +304,19 @@ function Install-OpenCover
     
     import-module Microsoft.PowerShell.Archive
-    Expand-ZipArchive -Path "$env:TEMP/$filename" -DestinationPath "$TargetDirectory/OpenCover"
+    Expand-Archive -Path $tempPath -DestinationPath "$TargetDirectory/OpenCover"
```
>Expand-Archive [](start = 4, length = 14)

This should be Expand-ZipArchive as Expand-Archive is not available on PSv4. Azure functions does not support PSv5 yet.

SteveL-MSFT replied:
The archive module supports ps4, can we just install it from the gallery when using Azure functions?

adityapatwardhan commented on test/tools/OpenCover/OpenCover.psm1:
```diff
@@ -118,43 +124,9 @@ function Get-CoverageSummary([xml.xmlelement] $element)
-}
-
-function Expand-ZipArchive([string] $Path, [string] $DestinationPath)
-{
```
Re-instate this function as Expand-Archive is not available on PSv4.

SteveL-MSFT replied:
The archive module supports ps4, can we just install it from the gallery when using Azure functions?

adityapatwardhan replied:
I believe PSGet is not available on PSv4 so cannot install from gallery.

SteveL-MSFT replied:
PSGet supports v4, but isn't there by default.  Not a fan of duplicating expand-archive, but I guess we can keep it for now if expand-archive isn't available.

SteveL-MSFT committed:
addressing code review feedback
- check for git before using
- check for opencover.console in path
- fixed path to code coverage build of PS

SteveL-MSFT committed:
added Expand-ZipArchive back to support psv4 scenario

adityapatwardhan commented:
@SteveL-MSFT Are you missing any format files or type files? I see you added PSTypeNames.

SteveL-MSFT commented:
@adityapatwardhan in process of adding formatting, but wanted to get this merged in first.  rather than having the report as properties on the returned object, I want to have separate cmdlets for assembly coverage and the summary with independent formatting

adityapatwardhan commented:
LGTM.
